pdns/logger.cc

```diff
@@ -120,6 +120,9 @@ void Logger::log(const string &msg, Urgency u) noexcept
       catch (const runtime_error& e) {
         cerr << e.what() << endl;
       }
+      catch (boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<std::length_error> >& e2) {
```

Comment:
is there no less specific form we can catch? what throws this?

Reply:
` S.ringAccount("logmessages",msg);`

Reply:
```
1. fun_call_w_exception: Called function throws an exception of type boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<std::length_error> >.
 d_rings[name].account(item);
```

Reply:
ping @Habbie

Reply:
If you can put a comment in there explaining why we catch this very specific thing, it's fine by me.

Reply:
I think it is possible to happen if d_rings[name] does not exist, but I wonder if boost would let me catch std::length_error instead.

Reply:
IMO this is too specific: if boost changes an implementation detail this won't match.

Reply:
removed as impossible.